### ISSUE TITLE
Running ispc on Windows with --target=aarch64

### ISSUE THREAD

**Issue by steplong** [ispc/ispc#2379]
When I run ISPC on Windows with --target=aarch64, it says "Error: Aarch64 is not supported on Windows.". Is there any plans to add support for Aarch64 on Windows?

**Comment by dbabokin** [comment]
Yes, there are plans to add that.

What is expected hardware target for this support in your case? Window ARM is still quite a rare beast and I'm a bit confused what hardware flavor we should target (and test). Figuring out this will help moving forward.

**Comment by dbabokin** [comment]
@steplong also patches and PRs are welcome.

**Comment by steplong** [comment]
We are looking to target Windows ARM64 snapdragon laptops

**Comment by steplong** [comment]
> @steplong also patches and PRs are welcome.

 What is necessary to add support for Aarch64 on Windows?

**Comment by dbabokin** [comment]
@steplong I asked folks who were experimenting with Windows ARM support to reach out to you privately. If you still have questions / need guidance how to enable Windows ARM target - please let me know.

**Comment by steplong** [comment]
Hi, I'm having trouble building ispc on Windows with https://github.com/ispc/ispc/pull/2381. The commands I used are below:
```
# Build LLVM 13.0
python alloy.py -b --version=13.0
cmake -G "Visual Studio 17 2022" -Thost=x64 -DCMAKE_INSTALL_PREFIX="PATH/TO/INSTALL" -DISPC_CROSS=ON \
           -DISPC_GNUWIN32_PATH="PATH/TO/GNUWIN32" -DISPC_LINUX_TARGET=OFF -DISPC_FREEBSD_TARGET=OFF \
           -DISPC_MACOS_TARGET=OFF -DISPC_ANDROID_TARGET=OFF -DISPC_IOS_TARGET=OFF -DISPC_PS_TARGET=OFF ..
msbuild INSTALL.vcxproj /V:m /p:Platform=x64 /p:Configuration=Release -t:rebuild
```

First, I'm getting the following error, but the build seemed to progress regardless. Maybe, this is just a warning and doesn't break the build. I'm using the m4 provided here https://gnuwin32.sourceforge.net/packages/m4.htm.
```
MSBuild version 17.3.1+2badb37d1 for .NET Framework
Checking Build System
Generating Release/builtins-dispatch.cpp
path\to\m4.exe: internal error detected; please report this bug to <bug-m4@gnu.org>: Aborted
Generating Release/builtins-dispatch-macos.cpp
path\to\m4.exe: internal error detected; please report this bug to <bug-m4@gnu.org>: Aborted
Generating Release/builtins-target-sse2-i32x4-32bit-windows.cpp
path\to\m4.exe: internal error detected; please report this bug to <bug-m4@gnu.org>: Aborted
Generating Release/builtins-target-sse2-i32x4-32bit-unix.cpp
path\to\m4.exe: internal error detected; please report this bug to <bug-m4@gnu.org>: Aborted
Generating Release/builtins-target-sse2-i32x4-64bit-windows.cpp
path\to\m4.exe: internal error detected; please report this bug to <bug-m4@gnu.org>: Aborted
```
I'm getting the following error when building ispc_cpu_examples:
```
Generating ao_ispc.h, ao.ispc.obj, ao_ispc_sse2.h, ao.ispc_sse2.obj, ao_ispc_sse4.h, ao.ispc_sse4.obj,
ao_ispc_avx.h, ao.ispc_avx.obj, ao_ispc_avx2.h, ao.ispc_avx2.obj, ao_ispc_avx512knl.h, ao.ispc_avx512knl.obj,
ao_ispc_avx512skx.h, ao.ispc_avx512skx.obj
path\to\ispc\src\builtins.cpp:1121: Assertion failed: "func != NULL".
***
*** Please file a bug report at https://github.com/ispc/ispc/issues
*** (Including as much information as you can about how to reproduce this error).
*** You have apparently encountered a bug in the compiler that we'd like to fix!
***
```
When I run the ispc.exe I built with --target=neon-i32x8, I get:
```
Error: neon-i32x8 target for aarch64 on Windows is not supported in current build.
```

**Comment by dbabokin** [comment]
Yeah, the changes are not complete. I'll iterate on the PR to make it work.

**Comment by steplong** [comment]
Should be resolved by #2381
